hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java

```diff
@@ -5457,7 +5421,11 @@ public boolean hasPathCapability(final Path path, final String capability)
     case STORE_CAPABILITY_DIRECTORY_MARKER_AWARE:
       return true;
 
-      // multi object delete flag
+    // this is always true, even if multi object
+    // delete is disabled -the page size is simply reduced to 1.
+    case CommonPathCapabilities.BULK_DELETE:
```

Comment:
it means the API is present and some of the semantics "parent dir existence not guaranteed". For that reason, it will always be faster than before: one DELETE; no LIST/HEAD etc